.github/workflows/clang_tidy.yml

```diff
@@ -0,0 +1,157 @@
+name: Clang-tidy static analysis (Ubuntu 24.04, Python 3.12, Clang-18, Clang-tidy-18)
```

Comment:
Please add this workflow name to the workflow rerunner, https://github.com/openvinotoolkit/openvino/blob/feffa0c210c65b2964941d10068e0fd876667c9e/.github/workflows/workflow_rerunner.yml#L5
so that if it fails because of a known sporadic issue, it is automatically restarted.

Reply:
Done

Reply:
@akashchi Rerunner test started to fail after the change

Reply:
This failure is test-related, I'll fix it in a separate PR. It is a not-required workflow so this PR could be merged.